url47ebb27197_asirbu_CorrectiveEvolution_tgz-pJ8-eu_fbk_soa_evolution_engine_impl_test_ProblemToSMVFuncTestJ8/src/eu/fbk/soa/evolution/sts/impl/NusmvAction.java

```diff
@@ -0,0 +1,124 @@
+package eu.fbk.soa.evolution.sts.impl;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import org.apache.log4j.Logger;
+
+import eu.fbk.soa.evolution.sts.Action;
+
+
+
+public class NusmvAction implements Action {
+
+	static Logger logger = Logger.getLogger(NusmvAction.class);
+	
+	private static int nrActions = 1;
+	
+	private int index;
+	
+	private String name = "";
+	
+	private Map<String, String> variableValues;
+	
+	private boolean isInput = true;
+	
+	
+	/**
+	 * @throws IllegalArgumentException if all input/output actions are UNDEF
+	 */
+	public NusmvAction(Map<String, String> varValues) throws IllegalArgumentException {		
+		index = nrActions;
+		nrActions++;
+		variableValues = new HashMap<String, String>();
+				
+		String actionBody = "";
+		
+		for (String key : varValues.keySet()) {
+			if (key.contains("input") || key.contains("output")) {
+				String value = varValues.get(key);
+				
+				if (!value.contains("UNDEF")) {
+					variableValues.put(key, value);
+					actionBody += key + " = " + value + "\n";
+//					name = key + " = " + value;
+					name = value;
+					logger.trace("Created action: " + name);
+					if (key.contains("output")) {
+						this.isInput = false;
+					}
+				}
+			}
+		}
+		if (name == "") {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**: An equality operator (== or !=) is used to compare strings, which matches strings based on address.

**Fix**: Do content comparison using the equals() method to compare the values of the strings.